### PR TITLE
player has lifes and if he is hit by explosion, lifes decreases

### PR DESCRIPTION
**Pull request by Abi2737**

### CODE REVIEW

File: BasePlayer.cpp [hunk 1]
```diff
@@ -398,6 +401,39 @@ void BasePlayer::Stay()
 	isMoving = false;
 }
 
+void BasePlayer::OnLifeLost()
+{
+	lifes--;
+	if (lifes == 0)
+	{
+		OnDeath();
+		return;
```
Comment (Abi2737) [inline BasePlayer.cpp:410]: nit: `return` not needed

File: BasePlayer.cpp [hunk 2]
```diff
@@ -398,6 +401,39 @@ void BasePlayer::Stay()
...
+	}
+}
+
+void BasePlayer::OnDeath()
+{
+	cout << name << " e mort" << endl;
```
Comment (Abi2737) [inline BasePlayer.cpp]: nit: romengleza :P

File: BasePlayer.cpp [hunk 3]
```diff
@@ -398,6 +401,39 @@ void BasePlayer::Stay()
...
+	return;
```
Comment (Abi2737) [inline BasePlayer.cpp]: nit: redundant `return`

File: Bomb.cpp [hunk 4]
```diff
@@ -435,9 +435,8 @@ void Bomb::MarkExplosionXSideInMap(int lenght, bool right, char ch)
 {
 	int sign = right ? 1 : -1;
 	MatPos pos = matPos;
-	pos.l += 1 * sign;
 
-	while (lenght)
+	while (lenght >= 0)
```
Comment (Abi2737) [inline Bomb.cpp:479]: nice catch ;) 👏

File: Constants.h [hunk 5]
```diff
@@ -39,6 +39,7 @@ namespace PlayerConst
 	constexpr float SPEED = 100;
 	constexpr int BOMB_COUNT = 3;
 	constexpr int BOMB_LENGTH = 3;
+	constexpr int LIFES = 3;
```
Comment (Abi2737) [inline Constants.h]: Thoughts  on naming it `LIFES_COUNT`?
And also, we could rename `BOX_COUNT` to `BOXES_COUNT` and `BOMB_COUNT` to `BOMBs_COUNT`

File: Bomb.cpp (hunk 4 above)
Comment (MateiEne) [reply Bomb.cpp:479]: haha :)) i m a genius :))

File: BasePlayer.h [hunk 6]
```diff
@@ -48,15 +50,19 @@ class BasePlayer
 	);
 	void InitAnimation(Animation<sf::IntRect>& animation, const int count, const MatPos frames[]);
 
-	void UpdateMovement(float dt);
 
+	void UpdateMovement(float dt);
+	void HitBox();
 
 protected:
+	int lifes;
```
Comment (Abi2737) [inline BasePlayer.h]: would you like more: `numOfLifes`?

File: Bomb.cpp (hunk 4 above)
Comment (Abi2737) [reply Bomb.cpp:479]: :))) i think you're also the guy who wrote these methods

File: Constants.h (hunk 5 above)
Comment (MateiEne) [reply Constants.h]: done

File: Bomb.cpp (hunk 4 above)
Comment (MateiEne) [reply Bomb.cpp:479]: xD

File: BasePlayer.cpp (hunk 2 above)
Comment (MateiEne) [reply BasePlayer.cpp]: 📦 :P

File: BasePlayer.cpp (hunk 1 above)
Comment (MateiEne) [reply BasePlayer.cpp:410]: ok

**Pull request merged**